lib/net.c

```diff
@@ -1200,11 +1201,80 @@ start_iface(struct gatekeeper_if *iface, unsigned int num_attempts_link_get)
 	return ret;
 }
 
+static void
+calculate_net_configure_para(struct net_config *net_conf)
+{
+	int i;
+
+	/*
+	 * The number of elements in the mbuf pool.
+	 *
+	 * The optimum size (in terms of memory usage) for a mempool is when
+	 * it is a power of two minus one.
+	 *
+	 * Need to provision enough memory for the worst case,
+	 * since each queue needs at least
+	 * gatekeeper_num_rx_desc + gatekeeper_num_tx_desc +
+	 * gatekeeper_max_pkt_burst descriptors. i.e.,
+	 * GATEKEEPER_DESC_PER_QUEUE = (gatekeeper_num_rx_desc +
+	 * 	gatekeeper_num_tx_desc + gatekeeper_max_pkt_burst).
+	 *
+	 * So, the pool size should be at least the maximum number of queues *
+	 * number of descriptors per queue, i.e.,
+	 * (gatekeeper_num_ports * gatekeeper_num_queues *
+	 * 	GATEKEEPER_DESC_PER_QUEUE - 1).
+	 */
+	net_conf->gatekeeper_num_mbuf = rte_align32pow2(
+		net_conf->gatekeeper_num_ports *
+		net_conf->gatekeeper_num_queues * (
+		net_conf->gatekeeper_max_pkt_burst +
+		net_conf->gatekeeper_num_desc)) - 1;
```

Comment:
Perhaps a silly question, but doesn't `gatekeeper_num_mbuf` depend on the number of lcores since there'll be blocks reserved per lcore? If so, this likely affects `gatekeeper_per_core_cache_size` as well. If not, cloud you include the answer to this question to the comment in the code?